plasma_framework/python_tests/tests/contracts/root_chain/test_challenge_in_flight_exit_output_spent.py

```diff
@@ -21,6 +22,57 @@ def test_challenge_in_flight_exit_output_spent_should_succeed(testlang, period):
     assert not in_flight_exit.output_piggybacked(0)
 
 
+@pytest.mark.parametrize(
+    "ife_tx_num_outputs,double_spend_output_index,challenging_tx_num_inputs,double_spend_input_index",
+    [(i, j, k, l) for i in range(1, 5) for j in range(0, i) for k in range(1, 5) for l in range(0, k)]
+)
+def test_challenge_in_flight_exit_output_spent_should_succeed_for_all_indices(testlang,
+                                                                              ife_tx_num_outputs,
+                                                                              double_spend_output_index,
+                                                                              challenging_tx_num_inputs,
+                                                                              double_spend_input_index):
+    deposit_amount = 100
+    deposit_id = testlang.deposit(testlang.accounts[0], deposit_amount)
+
+    owners = []
+    outputs = []
+    tx_output_amount = deposit_amount // ife_tx_num_outputs
+    for i in range(0, ife_tx_num_outputs):
+        owners.append(testlang.accounts[i])
+        outputs.append((testlang.accounts[i].address, NULL_ADDRESS, tx_output_amount))
+
+    ife_tx_id = testlang.spend_utxo([deposit_id], owners, outputs=outputs)
+    blknum, tx_index, _ = decode_utxo_id(ife_tx_id)
+    double_spend_owner = owners[double_spend_output_index]
+    double_spend_utxo = encode_utxo_id(blknum, tx_index, double_spend_output_index)
+
+    testlang.start_in_flight_exit(ife_tx_id)
+    testlang.piggyback_in_flight_exit_output(ife_tx_id, double_spend_output_index, double_spend_owner)
+
+    inputs = []
+    for i in range(0, challenging_tx_num_inputs):
+        if i == double_spend_input_index:
+            inputs.append(double_spend_utxo)
+        else:
+            inputs.append(testlang.deposit(double_spend_owner, tx_output_amount))
+
+    challenge_tx_id = testlang.spend_utxo(
+        inputs,
+        [double_spend_owner for i in range(0, challenging_tx_num_inputs)],
+        [
+            (double_spend_owner.address, NULL_ADDRESS, tx_output_amount),
+        ],
+        force_invalid=True
+    )
+
+    testlang.challenge_in_flight_exit_output_spent(ife_tx_id, challenge_tx_id, double_spend_output_index, double_spend_owner)
+
+    in_flight_exit = testlang.get_in_flight_exit(ife_tx_id)
+    MAX_INDEX_SIZE = 4
```

Comment:
Please put constant definitions on top of the file.

Reply:
yah, was copy-pasting from #593. I think this needs to be cleaned in bulk there, after this PR-to-a-PR lands